src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
lets add a bound, to prevent from creating for example a MachineProducer<String>